src/main/java/io/debezium/connector/cassandra/SchemaHolder.java

```diff
@@ -5,154 +5,305 @@
  */
 package io.debezium.connector.cassandra;
 
-import static java.util.stream.Collectors.toList;
-
-import java.util.Collection;
+import java.lang.reflect.Method;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 
-import org.apache.cassandra.db.marshal.AbstractType;
-import org.apache.kafka.connect.data.Schema;
-import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.cassandra.config.CFMetaData;
+import org.apache.cassandra.config.Schema;
+import org.apache.cassandra.db.ColumnFamilyStore;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.schema.KeyspaceParams;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.datastax.driver.core.ColumnMetadata;
+import com.datastax.driver.core.AggregateMetadata;
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.FunctionMetadata;
 import com.datastax.driver.core.KeyspaceMetadata;
+import com.datastax.driver.core.MaterializedViewMetadata;
+import com.datastax.driver.core.SchemaChangeListener;
 import com.datastax.driver.core.TableMetadata;
+import com.datastax.driver.core.UserType;
 
 import io.debezium.connector.SourceInfoStructMaker;
-import io.debezium.connector.cassandra.exceptions.CassandraConnectorSchemaException;
-import io.debezium.connector.cassandra.transforms.CassandraTypeConverter;
-import io.debezium.connector.cassandra.transforms.CassandraTypeDeserializer;
 
 /**
- * Caches the key and value schema for all CDC-enabled tables. This cache gets updated
- * by {@link SchemaProcessor} periodically.
+ * Listening to schema changes in Cassandra DB and caches the key and value schema for all CDC-enabled tables.
+ * This cache gets updated whenever there's a schema change in Cassandra DB
```

Comment:
A general question on the handling of schema changes; IIUC, the connector is triggered via a listener in case there's a schema change; how is this sync-ed with the log reading process? E.g. when reading changes from a log file created _before_ that schema change, how are we making sure that those parsed events are propagated with the previous, then valid schema?

Reply:
This is not related to this PR, but related to PR#26. I left a comment in DBZ-3417 to answer this question and explain about the trade off: https://issues.redhat.com/browse/DBZ-3417?focusedCommentId=16227506&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16227506

Reply:
> A general question on the handling of schema changes; IIUC, the connector is triggered via a listener in case there's a schema change; how is this sync-ed with the log reading process? E.g. when reading changes from a log file created _before_ that schema change, how are we making sure that those parsed events are propagated with the previous, then valid schema?

This is not related to this PR, but related to PR#26. I left a comment in DBZ-3417 to answer this question and explain about the trade off in detail: https://issues.redhat.com/browse/DBZ-3417?focusedCommentId=16227506&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16227506

Reply:
Ok, thx! Also thx a lot for explaining this to me in the call yesterday!